docs/src/references/child-devices.md

```diff
@@ -0,0 +1,151 @@
+
+```json
+{
+    "config-types": ["data-mapping-config", "child-device-mapping-config"]
+}
+```
+
+* The config list can also be requested explicitly by publishing a request to `tedge/commands/req/config/list` topic
+* All processes that supports config management must respond to these requests by sending their config-list to `tedge/commands/res/config/list/<device_id>` topic
+* The mapper then combines the config list published by different components and updates the same in the cloud
+
+#### Pull config snapshot from device
+
+* The mapper sends the config snapshot request from the cloud to the child device specific topic, with config type
+  E.g: `tedge/commands/req/config/snapshot/<config_type>/<device_id>`
+* The child device process listening on these snapshot requests acknowledges the receipt of this request by publishing an `executing` status to `tedge/commands/req/config/snapshot/<config_type>/<device_id>`
+* The mapper, on receipt of this `executing` status forwards it to the connected cloud.
+* The child device process then uploads the config snapshot to thin-edge via the file upload REST APIs exposed by `tedge-agent`.
+  The URL for the uploaded binary is returned in the response by `tedge-agent`.
+* The child device process marks the operation successful by sending the `successful` status to `tedge/commands/req/config/snapshot/<config_type>/<device_id>` with the binary URL received from `tedge-agent`
+* The mapper on receipt of this successful message uses the URL in that response to download that file from `tedge-agent` and uploads it to the connected cloud and marks the operation successful in the cloud as well.
+
+#### Push config update to device
+
+* The mapper receives a config update request from the cloud with the config type and the config binary download URL
+* The mapper then downloads the config binary from that download URL in the request
+* The mapper then uploads this downloaded config file to thin-edge via the file upload REST APIs exposed by `tedge-agent`.
+  The URL for the uploaded binary is returned in the response by `tedge-agent`.
+* The mapper then sends the config update request to `tedge/commands/req/config/update/<config_type>/<device_id>` with the binary URL received from `tedge-agent` in the payload.
+* The child device process acknowledges the receipt of this update request by publishing an `executing` status to `tedge/commands/req/config/update/<config_type>/<device_id>`
+* The mapper, on receipt of this `executing` status forwards it to the connected cloud.
+* The child device process then downloads the config file from `tedge-agent` using the URL in the request and applies it.
+* The child device process marks the operation successful by sending the `successful` status to `tedge/commands/req/config/update/<config_type>/<device_id>`
+* The mapper on receipt of this message marks the operation successful in the cloud as well.
```

Comment:
That approach pushs a lot of effort to the child-device. Here basically the C8Y operation is just forwarded to the child and the child has to take care of whole operation handling. For industrial device as by example a PLC it will be hard or even impossible to use that kind of child-device interface.

From my POV we should solve much more of that in the plugin. And then the plugin should have a thin interface for the 3 aspects below towards the child-device:
  1) inerface to consume/provide config-files from/to external/device via network (e.g. MQTT)
  2) provide a mechanism to associate a config-file with a child-dev twin in the cloud
  3) provide a mechanism/interface to allow a child-device to provision it-self at thin-edge

Reply:
One more comment to the strucure: I prefer to have topics for certain functionalities in separate documents. For child-device support of the C8Y Config Plugin I have used the plugin's reference doc to add a user-faced description of my approach (see https://github.com/thin-edge/thin-edge.io/pull/1236).

Reply:
> That approach pushs a lot of effort to the child-device. Here basically the C8Y operation is mostly forwarded to the child and the child has to take care of whole operation handling, and has to deal with MQTT and HTTP. For industrial devices as by example a PLC it will be hard or even impossible to use that kind of child-device interface.

The case of PLCs and devices with low extensibility capabilities can be addressed once and for all. These devices must be connected via a device-specific process that translates the thin-edge protocol into the device-specific protocol.